### ISSUE TITLE
ci-flake: mz_cluster_replica_utilization is used in an SLT test while not being deterministic

### ISSUE THREAD

**Issue by philip-stoev** [MaterializeInc/materialize#16619]
### What version of Materialize are you using?

add23fd7259684d689a7cf4e5fbdc92466f21a00

### How did you install Materialize?

Docker image

### What is the issue?

The SLT has started failing as follows:

```
==> test/sqllogictest/cluster.slt
    SELECT replica_id, process_id, cpu_percent, memory_percent FROM mz_internal.mz_cluster_replica_utilization
    OutputFailure:test/sqllogictest/cluster.slt:553
            expected: Values(["1", "0", "0", "0", "2", "0", "0", "0", "3", "0", "0", "0", "23", "0", "0", "0", "24", "0", "0", "0", "25", "0", "0", "0", "25", "1", "0", "0"])
            actually: Values(["1", "0", "0", "0", "2", "0", "0", "0", "3", "0", "0", "0", "23", "0", "0", "0"])
            actual raw: [Row { columns: [Column { name: "replica_id", type: Other(Other { name: "uint8", oid: 16464, kind: Simple, schema: "pg_catalog" }) }, Column { name: "process_id", type: Other(Other { name: "uint8", oid: 16464, kind: Simple, schema: "pg_catalog" }) }, Column { name: "cpu_percent", type: Numeric }, Column { name: "memory_percent", type: Numeric }] }, Row { columns: [Column { name: "replica_id", type: Other(Other { name: "uint8", oid: 16464, kind: Simple, schema: "pg_catalog" }) }, Column { name: "process_id", type: Other(Other { name: "uint8", oid: 16464, kind: Simple, schema: "pg_catalog" }) }, Column { name: "cpu_percent", type: Numeric }, Column { name: "memory_percent", type: Numeric }] }, Row { columns: [Column { name: "replica_id", type: Other(Other { name: "uint8", oid: 16464, kind: Simple, schema: "pg_catalog" }) }, Column { name: "process_id", type: Other(Other { name: "uint8", oid: 16464, kind: Simple, schema: "pg_catalog" }) }, Column { name: "cpu_percent", type: Numeric }, Column { name: "memory_percent", type: Numeric }] }, Row { columns: [Column { name: "replica_id", type: Other(Other { name: "uint8", oid: 16464, kind: Simple, schema: "pg_catalog" }) }, Column { name: "process_id", type: Other(Other { name: "uint8", oid: 16464, kind: Simple, schema: "pg_catalog" }) }, Column { name: "cpu_percent", type: Numeric }, Column { name: "memory_percent", type: Numeric }] }]
    ----
    FAIL: output-failure=1 success=119 total=120
```

I believe the reason is that `mz_internal.mz_cluster_replica_utilization` is dependent on `mz_internal.mz_cluster_replica_metrics` which is not guaranteed to immediately have the expected data. I think this query should be a .td test so that testdrive can retry the query until the expected result is obtained. SLT does not retry queries. 

@umanwizard FYI

**Comment by lluki** [comment]
Having right now exactly the same problem in my draft PR https://github.com/MaterializeInc/materialize/pull/16588 .

**Comment by umanwizard** [comment]
Oops, you’re absolutely right. Will fix today.

**Comment by umanwizard** [comment]
Fixed by #16626
